File: app/src/main/java/com/dixitpatel/quidcodemo/dagger/modules/CommonAppModule.java
package com.dixitpatel.quidcodemo.dagger.modules;

import android.content.Context;

import com.dixitpatel.quidcodemo.application.MyApplication;
import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class CommonAppModule {

    @Singleton
    @Provides
    Context provideContext(MyApplication application) {
        return application;
    }

    @Singleton
    @Provides
    Gson provideGson() {
        return new Gson();
    }

}

File: app/src/main/java/com/dixitpatel/quidcodemo/dagger/modules/CommonAppModule.kt
package com.dixitpatel.quidcodemo.dagger.modules

import android.content.Context
import com.dixitpatel.quidcodemo.application.MyApplication
import com.google.gson.Gson
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class CommonAppModule {
    @Singleton
    @Provides
    fun provideContext(application: MyApplication): Context {
        return application
    }

    @Singleton
    @Provides
    fun provideGson(): Gson {
        return Gson()
    }
}